React/Modules/RCTTiming.m

```diff
@@ -181,22 +192,31 @@ - (void)didUpdateFrame:(__unused RCTFrameUpdate *)update
   // Call timers that need to be called
   if (timersToCall.count > 0) {
     [_bridge enqueueJSCall:@"JSTimersExecution.callTimers" args:@[timersToCall]];
-
-    // If we call at least one timer this frame, don't switch to a paused state yet, so if
-    // in response to this timer another timer is scheduled, we don't pause and unpause
-    // the displaylink frivolously.
-    return;
   }
 
-  // No need to call the pauseCallback as RCTDisplayLink will ask us about our paused
-  // status immediately after completing this call
-  if (_timers.count == 0) {
-    _paused = YES;
+  if (_sendIdleEvents) {
+    NSTimeInterval frameElapsed = (CACurrentMediaTime() - update.timestamp) * 1000;
+    NSTimeInterval currentTimestamp = [[NSDate date] timeIntervalSince1970] * 1000;
+    NSNumber *absoluteFrameStartMillis = [NSNumber numberWithDouble:currentTimestamp - frameElapsed];
```

Comment:
Also it's not a big deal but might be clearer to do the x1000 multiplication at the very end, so leave `frameElapsed` and `currentTimestamp` in seconds (which NSTimeInterval is supposed to be)